### PR TITLE
Fail fast for preview/guess mode for rate limit

### PR DESCRIPTION
**Pull request by vinhvd**
Context : 
When reaching rate limit, zendesk endpoint will return `retry_after`, the number of seconds to wait.  The retry is manual and forever until rate limit is released. It is necessary for preview/guess mode.

This PR is to improve the preview/guess to be fail fast for the kind of retry.

### CODE REVIEW

File: test/embulk/input/zendesk/test_client.rb [hunk 1]
```diff
@@ -482,6 +482,14 @@ def stub_response(status, headers = [], body_json = nil)
...
             mock(client).sleep after.to_i
             assert_throw(:retry) do
+              client.tickets(false, &proc{})
+            end
+          end
+
+          test "429 guess/preview fail fast" do
+            after = "123"
+            stub_response(429, ["Retry-After: #{after}"])
+            assert_raise(DataError.new("Rate Limited. Waiting #{after} seconds to re-run")) do
```
Comment (tvhung83) [inline test/embulk/input/zendesk/test_client.rb:492]: Can we assert that the method is executed only once, ie. no retry?

Comment (vinhvd) [reply test/embulk/input/zendesk/test_client.rb:492]: Data Error means no retry https://github.com/treasure-data/embulk-input-zendesk/blob/3e749f3fc9af1889312a77f06f616f0a95886326/lib/embulk/input/zendesk/client.rb#L241 , isn't it ?

Comment (tvhung83) [reply test/embulk/input/zendesk/test_client.rb:492]: It could be retried > gave up > and still `DataError`?

Comment (tvhung83) [reply test/embulk/input/zendesk/test_client.rb:492]: I overlooked the test above this test, so it's good enough if it doesn't throw `:retry` (though some comments would be helpful).

**Review by tvhung83 — APPROVED** [review]
LGTM

**Pull request merged**